packages/discord.js/typings/index.d.ts

```diff
@@ -1507,17 +1504,17 @@ export class Message<Cached extends boolean = boolean> extends Base {
   public system: boolean;
   public readonly thread: ThreadChannel | null;
   public tts: boolean;
-  public type: MessageTypeKey;
+  public type: MessageType;
```

Comment:
What’s `MessageType` (same for other types) and why can we use it without `keyof typeof`?

Reply:
`MessageType` is an enum from -types. I don't understand the second question, nothing should be using `keyof` anymore since djs doesn't accept strings directly.

Reply:
Why does it not?

Reply:
> This makes discord.js no longer use string representations of enums internally. As a result if strings are still desired to be used, you must use the an enum resolver

Reply:
I hadn’t read that, that’s very annoying
Couldn’t you use these resolvers whenever there’s user input to allow us to pass strings?

Reply:
No, we won't be using resolvers internally. You either will need to use the enum or the resolver.